src/main/c/Server.cpp

```diff
@@ -90,14 +90,15 @@ std::ostream& operator<<(std::ostream& o, const EventBits& b) {
 
 constexpr int EpollTimeoutMillis = 500; // Twice a second is ample.
 constexpr int DefaultLameConnectionTimeoutSeconds = 10;
-pid_t gettid() {
-    return static_cast<pid_t>(syscall(SYS_gettid));
-}
 
 }
 
 namespace seasocks {
 
+pid_t gettid() {
```

Comment:
Interesting! This must be new. On my machine (Ubuntu 19.04):

```
SYNOPSIS
       #include <sys/types.h>

       pid_t gettid(void);

       Note: There is no glibc wrapper for this system call; see NOTES.
```

This is why I wrote it (originally back in RH5 days). I'd be happy to `#ifdef` this if we can detect if the system has a `gettid()` but clearly that's still not everywhere. Either way works!

Reply:
To be honest, at first I glanced over the no glibc wrapper warning in my man page on bionic. Did some more reading and I now understand why the wrapper was necessary and learned about `syscall` :+1: 

The man page on Eoan does state:
```
SYNOPSIS
       #include <sys/types.h>

       pid_t gettid(void);

DESCRIPTION
       gettid()  returns the caller's thread ID (TID).  In a single-threaded process, the thread ID is equal to the process ID (PID, as returned by getpid(2)).  In a multithreaded process, all threads have the same PID, but each one
       has a unique TID.  For further details, see the discussion of CLONE_THREAD in clone(2).
<snip>
VERSIONS
       The gettid() system call first appeared on Linux in kernel 2.4.11.  Library support was added in glibc 2.30.  (Earlier glibc versions did not provide a wrapper for this system call, necessitating the use of syscall(2).)
```
So this has indeed changed. We are definitely not the only ones running into this change. [This](https://www.spinics.net/lists/selinux/msg26619.html) thread on the selinux mailing list discusses it. Their resolution is indeed not to define the wrapper if glibc provides it.

I don't see the need for the `__BIONIC__` define they have, so I think it can be reduced to;
```cpp
#if !defined(__GLIBC_PREREQ)
  #define OVERRIDE_GETTID 1
#elif !__GLIBC_PREREQ(2,30)
  #define OVERRIDE_GETTID 1
#else
  #define OVERRIDE_GETTID 0
#endif
#if OVERRIDE_GETTID == 1
pid_t gettid() {
    return static_cast<pid_t>(syscall(SYS_gettid));
}
#endif
```
It's still very wordy and un-elegant if you ask me. I'm in favour of the simple and readable code that's currently on this PR. If you want me to change over to this approach with the preprocessor let me know and I'll implement it. Otherwise, lets merge this as is.

Reply:
I'm in total agreement: merging! :) thanks!